### PR TITLE
Add onNewConnection that is called when a new connection is created.

### PR DESCRIPTION
**Pull request by dreid**

### CODE REVIEW

**Comment by glyph** [comment]
Ticket: https://twistedmatrix.com/trac/ticket/8375

**Review by wsanchez — APPROVED** [review]
+1

**Comment by wsanchez** [comment]
Needs a topfile and has conflicts, but looks fine otherwise

File: twisted/application/test/test_internet.py [hunk 1]
```diff
@@ -810,3 +810,57 @@ def test_whenConnectedStopService(self):
...
+
+    def test_onNewConnectionCalledWithProtocol(self):
+      """
+      The C{onNewConnection} callable is passed the connected protocol instance.
+      """
+
+      class NewProtocol(Protocol, object):
+        """
+        A protocol for new connections.
+        """
```
Comment (twm) [inline twisted/application/test/test_internet.py:834]: The indentation here appears to be two spaces instead of four. Same for the previous test method.

**Review by twm — COMMENTED** [review]
Would love to see this merged. If you don't have time, @dreid, I'd be happy to shepherd this in.

**Comment by glyph** [comment]
@twm - pretty sure dreid has retired from twisted maintenance:). Would you mind taking this over? Thanks!

**Comment by twm** [comment]
@glyph Absolutely! I will take a look next week.

**Pull request closed without merging**